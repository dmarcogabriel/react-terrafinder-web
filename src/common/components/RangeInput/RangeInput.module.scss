@import 'styles';

.rangeInput {
  > button {
    padding: $size-sm;
    background: $white;
    width: 100%;
    cursor: pointer;

    > p {
      color: $gray-light;
      margin: 0.5rem 0;
    }

    .value {
      display: flex;
      flex-direction: row;
      justify-content: space-between;
      align-items: center;
      border: 1px solid $gray-light;
      border-radius: 0.25rem;
      padding: 0.8rem 0.5rem;
    }
  }

  .buttons {
    display: flex;
    flex-direction: column-reverse;

    @media ($desktop) {
      margin-top: 1rem;
      flex-direction: row;
      justify-content: space-between;
    }

    .cancelButton {
      background: transparent;
      color: $blue;
      margin: 1rem 0;
    }

    .okButton {
      background: $blue;
      margin: 1rem 0;
    }
  }
}
