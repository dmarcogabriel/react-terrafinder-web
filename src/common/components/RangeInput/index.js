import React, { useState } from 'react';
import Modal from 'common/components/Modal';
import Input from 'common/components/Input';
import Button from 'common/components/atm/Button';
import classes from './RangeInput.module.scss';

export default function RangeInput({
  label,
  ligature = 'até',
  value = [0, 1],
  onChange,
  dataTestId,
  valueDataTestId,
}) {
  const [showModal, setShowModal] = useState(false);
  const [minValue, setMinValue] = useState(value[0]);
  const [maxValue, setMaxValue] = useState(value[1]);

  const handleClose = () => setShowModal(false);

  const handleOk = () => {
    setShowModal(false);
    onChange([minValue, maxValue]);
  };

  return (
    <div className={classes.rangeInput}>
      <button
        data-testid={dataTestId || 'rangeInput'}
        type="button"
        className={classes.input}
        onClick={() => setShowModal(!showModal)}
      >
        <p>{label}</p>

        <div data-testid={valueDataTestId || 'value'} className={classes.value}>
          {`${minValue} ${ligature} ${maxValue}`}
        </div>
      </button>

      <Modal show={showModal}>
        <Input
          dataTestId="minInput"
          label="Mínimo"
          value={minValue}
          onChange={setMinValue}
        />

        <Input
          dataTestId="maxInput"
          label="Maximo"
          value={maxValue}
          onChange={setMaxValue}
        />

        <div className={classes.buttons}>
          <Button
            dataTestId="cancelButton"
            className={classes.cancelButton}
            onClick={handleClose}
          >
            Cancelar
          </Button>

          <Button
            dataTestId="okButton"
            className={classes.okButton}
            onClick={handleOk}
          >
            Ok
          </Button>
        </div>
      </Modal>
    </div>
  );
}
