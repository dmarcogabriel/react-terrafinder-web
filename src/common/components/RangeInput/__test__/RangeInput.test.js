import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
// import { render } from '@testing-library/react';
import RangeInput from '..';

describe('<RangeInput />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<RangeInput />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // Add tests here...

  it('matches snapshot', () => {
    const tree = renderer.create(<RangeInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
