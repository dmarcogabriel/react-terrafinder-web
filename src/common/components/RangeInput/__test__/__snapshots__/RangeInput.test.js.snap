// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RangeInput /> matches snapshot 1`] = `
<div>
  <button
    onClick={[Function]}
    type="button"
  >
    <p />
    <div>
      0 at√© 1
    </div>
  </button>
</div>
`;
